crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1553,12 +1553,13 @@ impl<'db> TypeInferenceBuilder<'db> {
             parenthesized: _,
         } = tuple;
 
-        for elt in elts {
-            self.infer_expression(elt);
-        }
+        let element_types = elts
+            .iter()
+            .map(|elt| self.infer_expression(elt))
+            .collect::<Vec<_>>();
 
         // TODO generic
```

Comment:
Seems like you did this ;)

```suggestion
```

Reply:
I think there's still types like `tuple[int, str]` or `tuple[int, ...]`? Maybe I should update the comment